service/controller/resource/values/create.go

```diff
@@ -17,10 +18,18 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		return microerror.Mask(err)
 	}
 
+	if app.GetDeletionTimestamp() != nil {
```

Comment:
That should be removed. `EnsureCreated` won't be called if the object is being deleted.